src/Stratis.Bitcoin/P2P/PeerAddress.cs

```diff
@@ -79,6 +79,45 @@ public IPAddress Loopback
         [JsonProperty(PropertyName = "lastSeen", NullValueHandling = NullValueHandling.Ignore)]
         public DateTime? LastSeen { get; private set; }
 
+        /// <summary>
+        /// UTC DateTime when a peer is banned.
+        /// </summary>
+        /// <remarks>
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
+        [JsonProperty(PropertyName = "bantimestamp", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTime? BanTimeStamp { get; set; }
+
+        /// <summary>
+        /// UTC DateTime when the ban expires against the peer.
+        /// </summary>
+        /// <remarks>
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
+        [JsonProperty(PropertyName = "banuntil", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTime? BanUntil { get; set; }
+
+        /// <summary>
+        /// Reason for banning the peer.
+        /// <remarks>
+        /// This is set in <see cref="PeerBanning"/>.
+        /// </remarks>
+        [JsonProperty(PropertyName = "banreason", NullValueHandling = NullValueHandling.Ignore)]
+        public string BanReason { get; set; }
```

Comment:
Did we decide we want to store the ban reason?

Reply:
Yes, I had a chat with @Aprogiena on this, and he said sizing wouldn't be an issue.